lib/charms/opencti/v0/opencti_connector.py

```diff
@@ -0,0 +1,237 @@
+            self.unit.status = ops.ActiveStatus()
+        except NotReady as exc:
+            self.unit.status = ops.WaitingStatus(str(exc))
+
+    def _reconcile_integration(self) -> None:
+        """Reconcile the charm integrations."""
+        if self.unit.is_leader():
+            integration = self.model.get_relation("opencti-connector")
+            data = integration.data[self.app]
+            data.update(
+                {
+                    "connector_charm_name": self._charm_name,
+                    "connector_type": self.connector_type,
+                }
+            )
+            if "connector_id" not in data:
+                data["connector_id"] = str(uuid.uuid4())
+
+    def _gen_env(self) -> dict[str, str]:
+        """Generate environment variables for the opencti connector service.
+
+        Returns:
+            Environment variables.
+        """
+        integration = self.model.get_relation("opencti-connector")
+        integration_data = integration.data[integration.app]
+        opencti_url, opencti_token_id = (
+            integration_data.get("opencti_url"),
+            integration_data.get("opencti_token"),
+        )
+        if not opencti_url or not opencti_token_id:
+            raise NotReady("waiting for opencti-connector integration")
+        opencti_token_secret = self.model.get_secret(id=opencti_token_id)
+        opencti_token = opencti_token_secret.get_content(refresh=True)["token"]
+        environment = {
+            "OPENCTI_URL": opencti_url,
+            "OPENCTI_TOKEN": opencti_token,
+            "CONNECTOR_ID": integration.data[self.app]["connector_id"],
+            "CONNECTOR_NAME": self.app.name,
+            "CONNECTOR_TYPE": self.connector_type,
+        }
+        for config, config_meta in self._config_metadata().items():
+            value = self.config.get(config)
+            if value is None:
+                continue
+            if self.boolean_style == "json" and isinstance(value, bool):
+                environment[self.kebab_to_constant(config)] = str(value).lower()
+            else:
+                environment[self.kebab_to_constant(config)] = str(value)
+        http_proxy = os.environ.get("JUJU_CHARM_HTTP_PROXY")
+        https_proxy = os.environ.get("JUJU_CHARM_HTTPS_PROXY")
+        no_proxy = os.environ.get("JUJU_CHARM_NO_PROXY")
+        if http_proxy:
+            environment["HTTP_PROXY"] = http_proxy
+            environment["http_proxy"] = http_proxy
+        if https_proxy:
+            environment["HTTPS_PROXY"] = https_proxy
+            environment["https_proxy"] = https_proxy
+        no_proxy_list = no_proxy.split(",") if no_proxy else []
+        if http_proxy or https_proxy:
+            opencti_host = urllib.parse.urlparse(opencti_url).hostname
+            no_proxy_list.append(opencti_host)
+            environment["NO_PROXY"] = https_proxy
+            environment["no_proxy"] = https_proxy
+        return environment
```

Comment:
Could we please have some new lines in here to have a few logically similar blocks please?

Reply:
Added, thanks!